### PR TITLE
Features/adc service

### PR DESCRIPTION
**Pull request by alejandrogonzalvo**
This PR implements: 

- The whole ADC service.
- Updates the PIN operators to work correctly inside a map.

For testing of the ADC service I have checked these corner cases:
image

Appart from that, it has been checked that the 3 ADC work in circular multichannel mode at the same time.
image.

### CODE REVIEW

**Review by PabloGonzalezSegarra — CHANGES_REQUESTED** [review]
No subas el .cproject

**Review by PabloGonzalezSegarra — COMMENTED** [review]
Porque borras todo lo relacionado con el PWM?

File: Src/HALAL/Services/ADC/ADC.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		status = HAL_ADC_Start_DMA(adcChannel.adc, (uint32_t*)adc_buf1, sizeof(adc_buf1) / sizeof(uint16_t));
+	}
+	else if (adcChannel.adc == &hadc2){
+		status = HAL_ADC_Start_DMA(adcChannel.adc, (uint32_t*)adc_buf2, sizeof(adc_buf2) / sizeof(uint16_t));
+	}
+	else if (adcChannel.adc == &hadc3){
+		status = HAL_ADC_Start_DMA(adcChannel.adc, (uint32_t*)adc_buf3, sizeof(adc_buf3) / sizeof(uint16_t));
+	}
+
+	if (status != HAL_OK) { }//Error_Handler(); }
```
Comment (PabloGonzalezSegarra) [inline Src/HALAL/Services/ADC/ADC.cpp:89]:
```suggestion
	if (status != HAL_OK) { }//Error_Handler()
```

File: Src/HALAL/Services/ADC/ADC.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+void ADC::turn_off_adc(uint8_t id) {
+	if (!service_IDs.contains(id)) { return; }
+	Pin pin = service_IDs[id];
+	ADCchannel adcChannel = pin_adc_map[pin];
+
+	HAL_StatusTypeDef status = HAL_ADC_Stop_DMA(adcChannel.adc);
+	if (status != HAL_OK) { } //Error_Handler(); }
```
Comment (PabloGonzalezSegarra) [inline Src/HALAL/Services/ADC/ADC.cpp:98]:
```suggestion
	if (status != HAL_OK) { } //Error_Handler()
```

File: Src/HALAL/Services/ADC/ADC.cpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+optional<uint16_t> ADC::get_pin_value(uint8_t id) {
+	if (!service_IDs.contains(id)) { return {}; }
+	Pin pin = service_IDs[id];
+	if (!pin_adc_map.contains(pin)) { return {}; }
+
+	ADCchannel adcChannel = pin_adc_map[pin];
```
Comment (PabloGonzalezSegarra) [inline Src/HALAL/Services/ADC/ADC.cpp:106]: Esta mal el convenio de nombres.

**Review by PabloGonzalezSegarra — COMMENTED** [review]
Últimos cambios y te lo acepto.

**Review by PabloGonzalezSegarra — DISMISSED** [review]

**Review by PabloGonzalezSegarra — APPROVED** [review]
Ya esta bien.

**Review by merycuesta — APPROVED** [review]

File: Src/HALAL/Services/ADC/ADC.cpp [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	if (adc_channel.adc == &hadc1){
+		return adc_read1[adc_channel.rank];
+	}
+	else if (adc_channel.adc == &hadc2){
+		return adc_read2[adc_channel.rank];
+	}
+	else if (adc_channel.adc == &hadc3){
+		return adc_read3[adc_channel.rank];
+	}
+}
```
Comment (PabloGonzalezSegarra) [inline Src/HALAL/Services/ADC/ADC.cpp:113]:
```suggestion
}

```

**Pull request merged**